pkgs/development/libraries/clfft/default.nix

```
@@ -0,0 +1,34 @@
+{ stdenv, fetchFromGitHub, cmake, fftw, fftwFloat, boost166, opencl-clhpp, ocl-icd }:
+
+let
+  version = "2.12.2";
+in stdenv.mkDerivation {
+  pname = "clfft";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "clMathLibraries";
+    repo = "clFFT";
+    rev = "refs/tags/v${version}";
+    sha256 = "134vb6214hn00qy84m4djg4hqs6hw19gkp8d0wlq8gb9m3mfx7na";
+  };
+
+  sourceRoot = "source/src";
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ fftw fftwFloat boost166 opencl-clhpp ocl-icd ];
+
+  meta = with stdenv.lib; {
+    description = "Library containing FFT functions written in OpenCL";
+    longDescription = ''
+      clFFT is a software library containing FFT functions written in OpenCL.
+      In addition to GPU devices, the library also supports running on CPU devices to facilitate debugging and heterogeneous programming.
+    '';
+    license = licenses.asl20;
+    homepage = "http://clmathlibraries.github.io/clFFT/";
+    platforms = platforms.linux;
```

It seems to fail on aarch64 because of upstream build options specific to x86. So let's restrict it to :
```suggestion
    platforms = [ "i686-linux" "x86_64-linux" ];
```